doc/release/0.18.0-notes.rst

```diff
@@ -65,18 +70,124 @@ zero-phase low-pass FIR filter, and downsamples using `scipy.signal.upfirdn`.
 functions in `numpy`.
 
+Sparse matrices now have a `count_nonzero` method, which counts the number of
+nonzero elements in the matrix. Unlike `getnnz()` and ``nnz`` propety,
+which return the number of stored entries (the length of the data attribute),
+this method counts the actual number of non-zero entries in data.
+
+
+`scipy.optimize` improvements
+-----------------------------
+
+The implementation of Nelder-Mead minimization,
+`scipy.minimize(..., method="Nelder-Mead")`, obtained a new keyword,
+`initial_simplex`, which can be used to specify the initial simplex for the
+optimization process.
+
+Initial step size selection in CG and BFGS minimizers has been improved. We
+expect that this change will improve numeric stability of optimization in the
+vast majortiy of use cases. See pull request #5536 for details.
+
+Handling of infinite bounds in SLSQP optimization has been improved. We expect
+that this change will improve numeric stability of optimization in the vast
+majortiy of use cases. See pull request #6024 for details.
+
+A large suite of global optimization benchmarks has been added to 
+``scipy/benchmarks/go_benchmark_functions``. See pull request #4191 for details.
+
+
+`scipy.stats` improvements
+--------------------------
+
+Trapezoidal distribution has been implemented as `scipy.stats.trapz`.
+Skew normal distribution has been implemented as `scipy.stats.skewnorm`.
+Burr type XII distribution has been implemented as `scipy.stats.burr12`. 
+
+Random matrices
+~~~~~~~~~~~~~~~
+
+`scipy.stats.special_ortho_group` and `scipy.stats.ortho_group` provide
+generators of random matrices in the SO(N) and O(N) groups, respectively. They
+generate matrices in the Haar distribution, the only uniform distribution on
+these group manifolds.
+
+`scipy.stats.random_correlation` provides a generator for random
+correlation matrices, given specified eigenvalues.
+
+
+`scipy.linalg` improvements
+---------------------------
+
+`scipy.linalg.svd` gained a new keyword argument, ``lapack_driver``. Available
+drivers are ``gesdd`` (default) and ``gesvd``.
+
+`scipy.linalg.lapack.ilaver` returns the version of the LAPACK library SciPy
+links to.
+
+
+`scipy.spatial` improvements
+----------------------------
+
+Boolean distances, `pdist`, has been sped up. Improvements vary by the function
+and the input size. In many cases, one can expect a speed-up of x2--x10. 
+
+New class `scipy.spatial.SphericalVoronoi` constructs Voronoi diagrams on the
+surface of a sphere. See pull request #5232 for details.
+
+`scipy.cluster` improvements
+----------------------------
+
+New clustering algorithm, the nearest neighbor chain algorithm, has been
+implemented for `scipy.cluster.hierarchy.linkage`. As a result, one can expect
+a significant algorithmic improvement for several linkage methods. 
+
+
+`scipy.special` improvements
+----------------------------
+
+`scipy.special.loggamma` computes the principal branch of the logarithm of the
```

Comment:
add that ``loggamma`` is new in 0.18.0. Preferably also with ``.. versionadded:: 0.18.0`` in the function docstring, I see that that's missing now.